jetty-server/src/main/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandler.java

```diff
@@ -0,0 +1,376 @@
+                        _aggregating = Boolean.FALSE;  // No body
+                    else
+                    {
+                        String ct = response.getContentType();
+                        if (ct == null)
+                            _aggregating = Boolean.TRUE;
+                        else
+                        {
+                            ct = MimeTypes.getContentTypeWithoutCharset(ct);
+                            _aggregating = isMimeTypeBufferable(StringUtil.asciiToLowerCase(ct));
+                        }
+                    }
+                }
+            }
+
+            // If we are not aggregating, then handle normally.
+            if (!_aggregating)
+            {
+                getNextInterceptor().write(content, last, callback);
+                return;
+            }
+
+            if (LOG.isDebugEnabled())
+                LOG.debug("{} aggregating", this);
+
+            try
+            {
+                aggregate(content);
+            }
+            catch (Throwable t)
+            {
+                resetBuffer();
+                callback.failed(t);
+                return;
+            }
+
+            if (last)
+                commit(callback);
+            else
+                callback.succeeded();
+        }
+
+        protected void aggregate(ByteBuffer content) throws IOException
+        {
+            if (_bufferedOutputStream == null)
+            {
+                // Create a new OutputStream to a file.
+                _file = Files.createTempFile(tempDir, "BufferedResponse", "").toFile();
+                _bufferedOutputStream = new BufferedOutputStream(Files.newOutputStream(_file.toPath(), StandardOpenOption.WRITE));
+            }
+
+            BufferUtil.writeTo(content, _bufferedOutputStream);
+        }
+
+        @Override
+        public Interceptor getNextInterceptor()
+        {
+            return _next;
+        }
+
+        protected void commit(Callback callback)
+        {
+            try
+            {
+                _bufferedOutputStream.flush();
+                _bufferedOutputStream.close();
+                _bufferedOutputStream = null;
+            }
+            catch (IOException e)
+            {
+                resetBuffer();
+                callback.failed(e);
+                return;
+            }
+
+            // Create an iterating callback to do the writing
+            IteratingCallback icb = new IteratingCallback()
+            {
+                private final long fileLength = _file.length();
+                private long _pos = 0;
+                private ByteBuffer _buffer;
+                private boolean _last = false;
+
+                @Override
+                protected Action process() throws Exception
+                {
+                    if (_last)
+                        return Action.SUCCEEDED;
+
+                    long len = Math.min(_outputBufferSize, fileLength - _pos);
```

Comment:
But output buffer size could be a few 10s of K and there is a lot of space between that and 2GB.   So you are getting all the cons of file mapped buffers with few of the benefits.   So as you say above, change it to:
```suggestion
                    long len = Math.min(Integer.MAX_VALUE, fileLength - _pos);
```